pkgs/build-support/trivial-builders.nix

```diff
@@ -2,11 +2,17 @@
 
 let
 
-  runCommand' = stdenv: name: env: buildCommand:
+  runCommand' = runLocal: stdenv: name: env: buildCommand:
     stdenv.mkDerivation ({
       inherit name buildCommand;
       passAsFile = [ "buildCommand" ];
-    } // env);
+    }
+    // env
+    // (if runLocal then {
+          preferLocalBuild = true;
+          allowSubstitutes = false;
+       } else {})
```

Comment:
I feel like this should be the other way around, such that `env` can override `preferLocalBuild`. Also could use `optionalAttrs runLocal { ... }` here

Reply:
You are absolutely right, see force push.

Tested manually with

```
nix-build -E 'with import ./. {}; runCommandLocal "foo" { x = "y"; preferLocalBuild = false; } "echo $x"'
```